src/render/SDL_render.c

```diff
@@ -4260,21 +4269,56 @@ SDL_RenderReadPixels(SDL_Renderer * renderer, const SDL_Rect * rect,
                                       format, pixels, pitch);
 }
 
+static void
+SDL_RenderSimulateVSync(SDL_Renderer * renderer)
+{
+    Uint32 now, elapsed;
+    const Uint32 interval = (1000 / 60); /* FIXME: What FPS? */
+
+    if (!interval) {
+        /* We can't do sub-ms delay, so just return here */
+        return;
+    }
+
+    now = SDL_GetTicks();
+    elapsed = (now - renderer->last_present);
+    if (elapsed < interval) {
+        Uint32 duration = (interval - elapsed);
+        SDL_Delay(duration);
+        now = SDL_GetTicks();
+    }
+
+    if (renderer->last_present) {
+        elapsed = (now - renderer->last_present);
```

Comment:
How does SDL2 behave in case of tick-overflow? Do you want to handle the case when `now` become less than `last_present`?

Reply:
That will only happen if it's days between the last time last_present was updated and now, and in that case you'll skip the delay and update last_present. I think it's fine.

Reply:
Oh, actually, I was thinking of the logic above. I'll fix that here.